lib/srv/app/gcp/handler.go

```diff
@@ -164,7 +169,10 @@ func (s *handler) serveHTTP(w http.ResponseWriter, req *http.Request) error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	s.Log.Debugf("Processing request, sessionId = %q, gcpServiceAccount = %q", sessionCtx.Identity.RouteToApp.SessionID, sessionCtx.Identity.RouteToApp.GCPServiceAccount)
+	s.Log.With(
+		"session_id", sessionCtx.Identity.RouteToApp.SessionID,
+		"gcp_service_account", sessionCtx.Identity.RouteToApp.GCPServiceAccount,
+	).DebugContext(context.Background(), "Processing request")
```

Comment:
You should be able to safely use `req.Context()` here rather than using `context.Background()`. This may even be preferred if we decide to automatically log request IDs in the future.

Slog discourages/prohibits log handlers from using the `context.Context` passed in for cancellation:

```
	// It is present solely to provide Handlers access to the context's values.
	// Canceling the context should not affect record processing.
	// (Among other things, log messages may be necessary to debug a
	// cancellation-related problem.)
```